onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -92,15 +107,16 @@ Status ScatterNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) co
 
   std::vector<int64_t> element_counts(last_indice_dimension, 0LL);  // Number of elements for each input dimension
 
+  TensorPitches input_strides(input_shape);
   for (int64_t i = 0; i < last_indice_dimension; ++i) {
-    element_counts[i] = input_shape.SizeFromDimension(i + 1);
```

Comment:
Clean up: Rather than calling SizeFromDimension() multiple times - compute strides once and use the componens needed